src/dsl/errors.ts

```diff
@@ -0,0 +1,31 @@
+export abstract class SingleError extends Error {
+  constructor(public message: string) {
+    super(message);
+  }
+
+  toString() {
+    return this.message;
+  }
+}
+
+export class UnimplementedSingleError extends SingleError {
+  constructor(featureName: string) {
+    super(`${featureName} is not implemented`);
+    this.name = "UnimplementedError";
+  }
+}
+
+abstract class MultipleError<T = SingleError> extends Error {
+  constructor(public errors: Array<T>) {
+    super(
+      `${errors.length} error${errors.length > 1 ? "s" : ""} occurred:\n\t* ${errors.join("\n\t* ")}\n\n`,
+    );
+    this.errors = errors;
+  }
+
+  toString() {
+    return this.message;
+  }
+}
+
+export class UnimplementedError extends MultipleError<UnimplementedSingleError> {}
```

Comment:
에러 네이밍이 조금 더 명시적이면 좋을거같아서 아래와 같이 수정해봤어요~

- Single, Multiple 개념이 헷갈려서 이렇게 정리하면 좋을거같아요.
  - Single: 생략
  - Multiple: "Multiple" prefix 붙이기
- `toString()` 구현이 들어간걸로 보이는데, 기본적으로 `Error` 객체에 구현되어있지 않나요? 따로 override하신 이유가 궁금해요.

```suggestion
enum ErrorCode {
  UNIMPLEMENTED = "UNIMPLEMENTED",
}

export class UnimplementedError extends Error {
  code = ErrorCode.UNIMPLEMENTED;
  
  constructor(featureName: string) {
    super(`${featureName} is not implemented`);
  }
}

export class MultipleError<T extends Error> extends Error {
  constructor(public errors: Array<T>) {
    super(
      `${errors.length} error${errors.length > 1 ? "s" : ""} occurred:\n\t* ${errors.join("\n\t* ")}\n\n`,
    );
  }

  toString() {
    return this.message;
  }
}

export class MultipleUnimplementedError extends MultipleError<UnimplementedError> {}
```

Reply:
> * `toString()` 구현이 들어간걸로 보이는데, 기본적으로 `Error` 객체에 구현되어있지 않나요? 따로 override하신 이유가 궁금해요.

fga/language 쪽 error 구현체를 가져오다보니 그런 것 같아요. 그 외에 제안주신 내용은 반영되었어요! [4fe1ce1](https://github.com/daangn/graplix/pull/9/commits/4fe1ce129d7a0801953cbfee868ccaee3c2897fc)